connectors/src/connectors/slack/temporal/activities.ts

```diff
@@ -602,3 +619,56 @@ export function formatDateForThreadTitle(date: Date) {
 
   return `${year}-${month}-${day}_${hours}h${minutes}`;
 }
+
+export async function getChannelsToGarbageCollect(
+  slackAccessToken: string,
+  connectorId: string
+) {
+  const remoteChannels = await getChannels(slackAccessToken);
+  const localChannels = await SlackMessages.findAll({
+    attributes: [
+      [Sequelize.fn("DISTINCT", Sequelize.col("channelId")), "channelId"],
+    ],
+    where: {
+      connectorId: connectorId,
+    },
+  });
+
+  const localChannelsIds = localChannels.map((c) => c.channelId);
+  const channelsToDeleteLocally = localChannelsIds.filter((lc) => {
+    return remoteChannels.find((rc) => rc.id === lc) === undefined;
```

Comment:
nit: you could use a set instead to have linear complexity

Reply:
Agree. I will fix it.